flink-core/src/main/java/org/apache/flink/api/java/typeutils/TupleTypeInfoBase.java

```diff
@@ -203,7 +205,26 @@ public void getFlatFields(String fieldExpression, int offset, List<FlatFieldDesc
 		TypeInformation<X> typed = (TypeInformation<X>) this.types[pos];
 		return typed;
 	}
-	
+
+	@Override
+	@PublicEvolving
+	public <F> FieldAccessor<T, F> getFieldAccessor(int pos, ExecutionConfig config) {
+		return new FieldAccessor.SimpleTupleFieldAccessor<T, F>(pos, this);
+	}
+
+	@Override
+	@PublicEvolving
+	public <F> FieldAccessor<T, F> getFieldAccessor(String fieldExpression, ExecutionConfig config) {
+		FieldAccessor.FieldExpressionDecomposition decomp = FieldAccessor.decomposeFieldExpression(fieldExpression);
+		int FieldPos = this.getFieldIndex(decomp.head);
```

Comment:
`getFieldIndex()` can return -1. This should be handled somewhere. I think the constructors of all accessors should check their input parameters (also for null etc.).